config/filter.d/screensharingd.conf

```diff
@@ -0,0 +1,33 @@
+# Fail2Ban configuration file
+#
+# Author: Simon Brown
+#
+# $Revision: 1 $ 
+#
+# Filter for Mac OS X Screen Sharing service
+
+[INCLUDES]
+
+# Read common prefixes. If any customizations available -- read them from
+# common.local
+before = common.conf
+
+
+[Definition]
+
+_daemon = screensharingd
+
+# Option:  failregex
+# Notes.:  regex to match the password failures messages in the logfile. The
+#          host must be matched by a group named "host". The tag "<HOST>" can
+#          be used for standard IP/hostname matching and is only an alias for
+#          (?:::f{4,6}:)?(?P<host>[\w\-.^_]+)
+# Values:  TEXT
+#
+failregex = ^.+ screensharingd.+: Authentication: FAILED :: User Name: .+ :: Viewer Address: <HOST> :: Type: DH$
```

Comment:
expression `^.+` makes no sense;
you should not restrict type to DH (`\w+` is absolutely sufficient)

Below the regexp that should work (untested)
```
failregex = ^%(__prefix_line)sAuthentication: FAILED :: User Name: .+ :: Viewer Address: <HOST> :: Type: \w+$
```